text/0079-cdk-2.0.md

```diff
@@ -0,0 +1,405 @@
+changes can however be introduced if a well documented proposal achieves consensus among the core development team. All
+breaking changes (including those planned in the prerelease announcement) will be documented in the release notes for
+each subsequent prerelease version, per the standard process.
+
+## Developer Preview
+
+At the end of the [Construct Library Maintenance Tasks](#construct-library-maintenance-tasks) window, a moratorium on
+breaking changes in v2 will be declared, as the codebase is thoroughly tested and vetted. At the end of the vetting
+period, and once known issues have been addressed, CDK v2 can be declared _Developer Preview_.
+
+Once the _Developer Preview_ has been declared, breaking changes will only be allowed if they are motivated by
+overwhelming customer feedback or the absence of alternate solutions to fix a well-documented pain point introduced by
+v2 changes.
+
+## Feedback Phase
+
+During _Developer Preview_, we will keep listening to customer feedback and address bugs as they are discovered. The CDK
+[version reporting] dataset will be used to track adoption of the new version. We will encourage customers to provide
+feedback on the upcoming new release using a pinned GitHub issue, and possibly other means.
+
+[version reporting]: https://docs.aws.amazon.com/cdk/latest/guide/tools.html
+
+## General Availability
+
+After the _Developer Preview_ has been sufficiently vetted, a timeline for _General Availability_ will be agreed upon by
+maintainers. The _General Availability_ checkist includes the following items:
+
+- [ ] Documentation
+  - [ ] API Reference Documentation
+  - [ ] User Guide
+  - [ ] Migration Guide
+  - [ ] Final maintenance plan for v1
+- [ ] Quality Assurance
+  - [ ] Complete suite of integration tests
+  - [ ] Examples built and tested against v2
+- [ ] No v1 feature is missing from v2 (unless it was deprecated in v1)
+
+Once the criteria have been satisfied, the high-level procedure for officially releasing CDK v2 is:
+
+1. Create a new `v1` branch from the `master` branch
+   1. Configure GitHub branch protection on `v1` as appropriate
+   1. Deploy CI/CD infrastructure on top of `v1`
+1. Make `next` the new `master` branch:
+   1. Force-push `next` on `master`
+   1. Delete `next` and all related CI/CD infrastructure
+1. Perform a _normal_ release
+1. Ceremoniously annouce CDK v2 is _Generally Available_, and repeat the v1 maintenance plan
+
+## Clean Up
+
+Now that the new major version has been declared _Generally Available_, the previous major (`v1`) has entered
+_maintenance_ mode. Forward porting is no longer exercised: instead, bug fixes on `master` will be back-ported to `v1`
+when necessary. This means `master` can diverge more significantly from `v1` than in earler stages of the process.
+
+The definition of `v1` feature flags and the code paths supporthing the _old behavior_ of those can be deleted.
```

Comment:
No the feature flags can be *neutralized* in `v2` (i.e: you CANNOT disable them), but the code paths must be kept in order to avoid tricky merge conflicts.

Reply:
Okay got it. Makes sense. It wasn't super clear to me what "neutralized" means.